astropy/coordinates/stokes_coord.py

```diff
@@ -0,0 +1,160 @@
+from collections import namedtuple
+from contextlib import contextmanager
+from copy import copy
+from typing import Dict
+
+import numpy as np
+
+from astropy.utils import unbroadcast
+from astropy.utils.shapes import ShapedLikeNDArray
+
+__all__ = ["StokesCoord", "custom_stokes_symbol_mapping", "StokesSymbol"]
+
+
+StokesSymbol = namedtuple("StokesSymbol", ["symbol", "description"], defaults=[""])
+
+# This is table 29 in the FITS 4.0 paper
+FITS_STOKES_VALUE_SYMBOL_MAP = {
+    1: StokesSymbol("I", "Standard Stokes unpolarized"),
+    2: StokesSymbol("Q", "Standard Stokes linear"),
+    3: StokesSymbol("U", "Standard Stokes linear"),
+    4: StokesSymbol("V", "Standard Stokes circular"),
+    -1: StokesSymbol("RR", "Right-right circular"),
+    -2: StokesSymbol("LL", "Left-left circular"),
+    -3: StokesSymbol("RL", "Right-left cross-circular"),
+    -4: StokesSymbol("LR", "Left-right cross-circular"),
+    -5: StokesSymbol("XX", "X parallel linear"),
+    -6: StokesSymbol("YY", "Y parallel linear"),
+    -7: StokesSymbol("XY", "XY cross linear"),
+    -8: StokesSymbol("YX", "YX cross linear"),
+}
+
+STOKES_VALUE_SYMBOL_MAP = copy(FITS_STOKES_VALUE_SYMBOL_MAP)
+
+
+@contextmanager
+def custom_stokes_symbol_mapping(
```

Comment:
I am not sure, this pattern is used for registering the frames with WCS implementations, it's really designed to be used by other packages like sunpy or radiospecta not users. Does that make it appropriate for `ScienceState`?

Reply:
I guess the reason here to try to make a decision is that it would be used as `[with ]stokes_symbol_mapping.set(mapping)`, i.e., have a slightly different API.

Reply:
There might be situations (hypothetically) where one wants to use context managers from two different packages to add two different sets of symbols, so essentially using nested context managers. I don't think ScienceState quite works that way - it feels like the ScienceState would *be* the mapping of symbols whereas the context manager @Cadair has here *extends* the mapping and can be used multiple times nested.

Reply:
I think that would not be difficult with `ScienceState` either, though you are right that is not the default. 

But I really mostly want to avoid introducing too many different APIs for similar things. How about following `astropy.units`? That has `{add|set}_enabled_{units|equivalencies|aliases}` - so, it could be `{add|set}_enabled_stokes_symbols` (which could of course use your function here underneath). Note that those can but do not have to be used as context managers; I think that may be useful here too (though perhaps already is! not sure exactly what `@contextmanager` does).

Anyway, do not want to make too big a deal of it, but obviously this is part of the public API, so it makes some sense to think about exactly how it should look.

Reply:
Hmm, if it existed already, then probably cleanest to just keep it as you have it...